construct/core.py

```diff
@@ -2263,7 +2307,11 @@ def _sizeof(self, context, path):
                 count = count(context)
         except (KeyError, AttributeError):
             raise SizeofError("cannot calculate size, key not found in context")
-        return count * self.subcon._sizeof(context, path)
+        size = 0
+        for i in range(count):
+            context._index = i
+            size += self.subcon._sizeof(context, path)
+        return size
```

Comment:
This should be a separate Issue entirely.

Reply:
Moved to new issue: 
https://github.com/construct/construct/issues/877